### PR TITLE
Ear bit fields cleaned

### PR DESCRIPTION
**Pull request by alan-stokes**
stuff to allow roberts ear to work correctly within pynn without hacks.

also adds support for static projections and will include selective recording. needs to be considered carefully for plastic projections.

needs prs:
https://github.com/SpiNNakerManchester/sPyNNaker8/pull/241
https://github.com/SpiNNakerManchester/sPyNNaker/pull/682
https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/459
https://github.com/SpiNNakerManchester/PACMAN/pull/236
https://github.com/SpiNNakerManchester/SpiNNMachine/pull/115
https://github.com/SpiNNakerManchester/DataSpecification/pull/69
https://github.com/SpiNNakerManchester/SpiNNUtils/pull/97
https://github.com/SpiNNakerManchester/SupportScripts/pull/18
https://github.com/SpiNNakerManchester/sPyNNaker8NewModelTemplate/pull/36
https://github.com/SpiNNakerManchester/PyNN8Examples/pull/27

### CODE REVIEW

**Comment by dkfellows** [comment]
Broken because the executable finder is throwing on no-such-APLX.

File: spinn_utilities/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+class SpiNNUtilsException(Exception):
```
Comment (dkfellows) [inline spinn_utilities/exceptions.py]: This class doesn't help at all. Just make `FailedToFindBinaryException` inherit from `Exception` directly.

File: spinn_utilities/executable_finder.py [hunk 2]
```diff
@@ -71,20 +73,22 @@ def get_executable_path(self, executable_name):
                 return potential_filename
 
         # No executable found
-        return None
+        raise FailedToFindBinaryException(
```
Comment (dkfellows) [inline spinn_utilities/executable_finder.py]: Is it worthwhile throwing here? Existing code expects a `None` return instead…

**Comment by Christian-B** [comment]
While this is nice work it is judged as too big an API change for the benefit.

Has also now fallen behind much of the other PRs based off the work here.

Closing PR but keeping branch for historical and future reference reasons.

**Pull request closed without merging**